tests/Repository/AnnouncementRepositoryTest.php

```diff
@@ -265,39 +271,131 @@ public function test_recurrence(): Void
     $announcement->setStartDate($announcement_start_date);
     $announcement->setEndDate($announcement_end_date);
     $announcement->setText('test_text');
+    $announcement->setApproval(1);
     $this->em->persist($announcement);
 
     $this->em->flush();
 
-    $announcement = $this->em
+    $post_announcement = $this->em
       ->getRepository(Announcement::class)
     ;
+    $post_count = array();
+    foreach (['-2 days', 'yesterday', 'now', 'tomorrow', '+2 days'] as $i)
+    {
+      $value = count($post_announcement->find_today($i));
+      array_push($post_count, $value);
+      
+    }
 
-    $this->assertSame(0, count($announcement->find_by_day('-2 days')));
+
+    $this->assertSame($pre_count[0], $post_count[0]);
     // ensure the announcement will not fire 2 days before
-    $this->assertSame(2, count($announcement->find_by_day('yesterday')));
-    $this->assertSame(6, count($announcement->find_by_day('now')));
-    $this->assertSame(2, count($announcement->find_by_day('tomorrow')));
-    $this->assertSame(0, count($announcement->find_by_day('+2 days')));
+    $this->assertSame($pre_count[1]+1, $post_count[1]);
+    $this->assertSame($pre_count[2]+1, $post_count[2]);
+    $this->assertSame($pre_count[3]+1, $post_count[3]);
+    $this->assertSame($pre_count[4], $post_count[4]);
     // ensure the announcement will not fire 2 days after
 
+  }
+
+  /**
+   * Testing announcement approval
+   * 
+   * Should create one unapproved announcement, then ensure that the pre count and post count haven't updated.
+   * This tests the database function and the find_today_approved() query function.
+   * 
+   * @author Daniel Boling
+   */
+  public function test_approval(): void
+  {
+    $test_user = $this->em
+    ->getRepository(User::class)
+    ->findOneByUsername("test_user");
+
+  $test_cat = $this->em
+    ->getRepository(Category::class)
+    ->findOneByName("test_category");
+
+  $pre_announcement_count = count($this->em
+    ->getRepository(Announcement::class)
+    ->find_today_approved())
+  ;
+
+  $announcement = new Announcement();
+  $announcement_start_date = new \DateTime('yesterday', new \DateTimeZone('GMT'));
+  $announcement_end_date = new \DateTime('tomorrow', new \DateTimeZone('GMT'));
+  $announcement->setSubject('test_subject');
+  $announcement->setAuthor('test_author');
+  $announcement->setCategory($test_cat);
+  $announcement->setUser($test_user);
+  $announcement->setStartDate($announcement_start_date);
+  $announcement->setEndDate($announcement_end_date);
+  $announcement->setText('test_text');
+  $announcement->setApproval(0);
+  $this->em->persist($announcement);
+
+  $this->em->flush();
+
+  $post_announcement_count = count($this->em
+    ->getRepository(Announcement::class)
+    ->find_today_approved())
+  ;
+  
+  $this->assertSame($pre_announcement_count, $post_announcement_count);
+  // these should both be the same, as the created announcement should not be approved.
```

Comment:
Could you add another test in here where you do a `find_today($approved=0)` and then assert that the count should be one more than the pre count?